### DISCUSSION TITLE
Level 9 metadata - does it matter?

### DISCUSSION THREAD

**Discussion by vpm-codes** [quietvoid/dovi_tool#96] [Q&A] [answered]
Hi,

Firstly, thank you for writing and sharing this tool, it works great.

My question is about DV level 9 metadata. Wikipedia mentions it is about mastering display colour primaries.

If I am generating a profile, should it include the level 9 metadata? What is the "default" for DV if it doesn't include this?

Specifically what I am doing is using the madmeasure dynamic optimiser on an HDR10 file & converting that to an RPU to play on an LG G1 with DV.

The HDR10 file says the mastering display colour primaries are DisplayP3 - should this be in the generator.json file somewhere? Would it affect anything either way?

**Comment by saindriches** [comment, marked as answer]
It doesn't really matter if the mastering display is P3, you can use source_primary_index = 0 for P3 sources.

**Reply by vpm-codes** [reply]
`source_primary_index = 0`

Thanks. This doesn't seem to make a difference to the output (the produced RPU hashes are the same) if I put it in the JSON as

`"source_primary_index": 0`

whether as its own line or under a "level9" section. Is it just unnecessary or am I not adding the option correctly?

**Reply by quietvoid** [reply]
Currently the metadata added to the JSON is ignored when generating from madVR, see #91.
I'm in the process of fixing it.

**Reply by quietvoid** [reply]
@vpm-codes This should now be possible in 1.4.3.

Example (for use with madVR measurements): 
```json5
{
    "level6": {
        "max_display_mastering_luminance": 1000,
        "min_display_mastering_luminance": 1,
        "max_content_light_level": 1000,
        "max_frame_average_light_level": 400
    },
    "default_metadata_blocks": [
        {
            "Level9": {
                "source_primary_index": 0
            }
        }
    ]
}
```

This does require `"cm_version": "V40"`, which is the default.
Otherwise the L9 metadata is not added.
